app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt

```diff
@@ -39,199 +35,57 @@ class RecentlyPlayedFragmentPresenter @Inject constructor(
   private lateinit var binding: RecentlyPlayedFragmentBinding
-  private lateinit var promotedStoryListAdapter: PromotedStoryListAdapter
-  private val itemList: MutableList<RecentlyPlayedItemViewModel> = ArrayList()
 
   fun handleCreateView(
     inflater: LayoutInflater,
     container: ViewGroup?,
-    internalProfileId: Int
+    profileId: Int
   ): View? {
-    binding = RecentlyPlayedFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
-
-    this.internalProfileId = internalProfileId
-
-    promotedStoryListAdapter = PromotedStoryListAdapter(itemList)
-    binding.ongoingStoryRecyclerView.apply {
-      adapter = promotedStoryListAdapter
-    }
-    binding.lifecycleOwner = fragment
-
-    subscribeToPromotedStoryList()
-    return binding.root
-  }
-
-  private val promotedStoryListSummaryResultLiveData:
-    LiveData<AsyncResult<PromotedActivityList>>
-    by lazy {
-      topicListController.getPromotedActivityList(
-        ProfileId.newBuilder().setInternalId(internalProfileId).build()
-      ).toLiveData()
-    }
-
-  private fun subscribeToPromotedStoryList() {
-    getAssumedSuccessfulPromotedActivityList().observe(
-      fragment,
-      {
-        if (it.promotedStoryList.recentlyPlayedStoryList.isNotEmpty()) {
-          addRecentlyPlayedStoryListSection(it.promotedStoryList.recentlyPlayedStoryList)
-        }
-
-        if (it.promotedStoryList.olderPlayedStoryList.isNotEmpty()) {
-          addOlderStoryListSection(it.promotedStoryList.olderPlayedStoryList)
-        }
-
-        if (it.promotedStoryList.suggestedStoryList.isNotEmpty()) {
-          addRecommendedStoryListSection(it.promotedStoryList.suggestedStoryList)
-        }
-
-        binding.ongoingStoryRecyclerView.layoutManager =
-          createLayoutManager(
-            it.promotedStoryList.recentlyPlayedStoryCount,
-            it.promotedStoryList.olderPlayedStoryCount,
-            it.promotedStoryList.suggestedStoryCount
-          )
-        promotedStoryListAdapter.notifyDataSetChanged()
-      }
-    )
-  }
-
-  private fun addRecentlyPlayedStoryListSection(
-    recentlyPlayedStoryList: MutableList<PromotedStory>
-  ) {
-    itemList.clear()
-    val recentSectionTitleViewModel =
-      SectionTitleViewModel(
-        resourceHandler.getStringInLocale(R.string.ongoing_story_last_week), false
-      )
-    itemList.add(recentSectionTitleViewModel)
-    recentlyPlayedStoryList.forEachIndexed { index, promotedStory ->
-      val ongoingStoryViewModel = createOngoingStoryViewModel(promotedStory, index)
-      itemList.add(ongoingStoryViewModel)
-    }
-  }
-
-  private fun createOngoingStoryViewModel(
-    promotedStory: PromotedStory,
-    index: Int
-  ): RecentlyPlayedItemViewModel {
-    return PromotedStoryViewModel(
-      activity,
-      promotedStory,
-      entityType,
+    this.internalProfileId = ProfileId.newBuilder().setInternalId(profileId).build()
```

Comment:
It looks like you're building another instance of ProfileId here, but I think you meant to assign the variable created earlier on line 46?

Reply:
@masclot, tagging you here since this is the comment I was referring to on chat.

Reply:
Nit: we also ought to call this ``profileId`` since it's not an "internal" profile ID anymore. Besides that, it should be fine for this variable to be a ``lateinit var``--no need to initialize it as @adhiamboperes suggests here (in case the most recent change was due to an old comment of mine--I'm not sure if that's the case here).